x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/MatchFunction.java

```diff
@@ -0,0 +1,132 @@
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isNotNull;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isString;
+
+/**
+ * Full text function that performs a {@link QueryStringQuery} .
+ */
+public class MatchFunction extends FullTextFunction implements Validatable {
+
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
+        Expression.class,
+        "Match",
+        MatchFunction::new
+    );
+
+    private final Expression field;
+
+    @FunctionInfo(
+        returnType = "boolean",
+        preview = true,
+        description = "Performs a match query on the specified field. Returns true if the provided query matches the row.",
+        examples = { @Example(file = "match-function", tag = "match-with-field") }
+    )
+    public MatchFunction(
+        Source source,
+        @Param(name = "field", type = { "keyword", "text" }, description = "Field that the query will target.") Expression field,
+        @Param(
+            name = "query",
+            type = { "keyword", "text" },
+            description = "Text you wish to find in the provided field."
+        ) Expression matchQuery
+    ) {
+        super(source, matchQuery, List.of(matchQuery, field));
+        this.field = field;
+    }
+
+    private MatchFunction(StreamInput in) throws IOException {
+        this(Source.readFrom((PlanStreamInput) in), in.readNamedWriteable(Expression.class), in.readNamedWriteable(Expression.class));
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        source().writeTo(out);
+        out.writeNamedWriteable(field);
+        out.writeNamedWriteable(query());
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
+    }
+
+    @Override
+    public String functionName() {
+        return "MATCH";
+    }
+
+    @Override
+    public Query asQuery(String queryText) {
+        return new MatchQuery(source(), ((FieldAttribute) field).name(), queryText);
+    }
+
+    @Override
+    protected TypeResolution resolveNonQueryParamTypes() {
+        return isNotNull(field, sourceText(), FIRST).and(isString(field, sourceText(), FIRST)).and(super.resolveNonQueryParamTypes());
+    }
+
+    @Override
+    public void validate(Failures failures) {
+        if (field instanceof FieldAttribute == false) {
+            failures.add(
+                Failure.fail(
+                    field,
+                    "[{}] cannot operate on [{}], which is not a field from an index mapping",
+                    functionName(),
+                    field.sourceText()
+                )
+            );
+        }
+    }
+
+    @Override
+    public Expression replaceChildren(List<Expression> newChildren) {
+        // Query is the first child, field is the second child
+        return new MatchFunction(source(), newChildren.get(1), newChildren.getFirst());
```

Comment:
We don't use `getFirst`, the majority of the functions call `newChildren.get(position)`. This is for the ESQL code to be consistent as much as possible in its entirety and not have multiple styles of writing code.

Reply:
OK, done in 2ddcbb597956e1597964ba9b7003d581f73881f2